frontend/javascripts/oxalis/controller/camera_controller.js

```diff
@@ -171,87 +173,64 @@ type TweenState = {
   upX: number,
   upY: number,
   upZ: number,
-  dx: number,
-  dy: number,
-  dz: number,
-  l: number,
-  r: number,
-  t: number,
-  b: number,
+  xPos: number,
+  yPos: number,
+  zPos: number,
+  left: number,
+  right: number,
+  top: number,
+  bottom: number,
 };
 
 export function rotate3DViewTo(id: OrthoView, animate: boolean = true): void {
   const state = Store.getState();
   const { dataset } = state;
-  const b = voxelToNm(dataset.dataSource.scale, getBoundaries(dataset).upperBoundary);
-  const pos = voxelToNm(dataset.dataSource.scale, getPosition(state.flycam));
-
-  const aspectRatio = getInputCatcherAspectRatio(state, OrthoViews.TDView);
+  const { tdCamera } = state.viewModeData.plane;
+  const flycamPos = voxelToNm(dataset.dataSource.scale, getPosition(state.flycam));
+  const datasetExtent = getDatasetExtentInLength(dataset);
   // This distance ensures that the 3D camera is so far "in the back" that all elements in the scene
   // are in front of it and thus visible.
   const clippingOffsetFactor = 900000;
-
-  let to: TweenState;
-  if (id === OrthoViews.TDView) {
-    const diagonal = Math.sqrt(b[0] * b[0] + b[1] * b[1]);
-    const padding = 0.05 * diagonal;
-
-    // Calculate the distance from (0, b[1]) in order to center the view
-    const a1 = b[0];
-    const b1 = -b[1];
-    const x1 = 0;
-    const y1 = b[1];
-    const x2 = pos[0];
-    const y2 = pos[1];
-
-    const b2 = 1 / Math.sqrt((b1 * b1) / a1 / a1 + 1);
-    const a2 = (-b2 * b1) / a1;
-    const d2 = ((a1 / b1) * (y1 - y2) - x1 + x2) / (-a2 + (a1 * b2) / b1);
-
-    const intersect = [x2 + d2 * a2, y2 + d2 * b2];
-    const distance = Dimensions.distance([x1, y1], intersect);
-
-    // Approximation to center the view vertically
-    const yOffset = pos[2] - b[2] / 2;
-
-    // Calulate the x coordinate so that the vector from the camera to the cube's middle point is
-    // perpendicular to the vector going from (0, b[1], 0) to (b[0], 0, 0).
-
-    const squareLeft = -distance - padding;
-    const squareRight = diagonal - distance + padding;
-    const squareTop = diagonal / 2 + padding + yOffset;
-    const squareBottom = -diagonal / 2 - padding + yOffset;
-    const squareCenterX = (squareLeft + squareRight) / 2;
-    const squareCenterY = (squareTop + squareBottom) / 2;
-    const squareWidth = Math.abs(squareLeft - squareRight);
-
-    const height = squareWidth / aspectRatio;
-
-    to = {
-      dx: (b[1] / diagonal) * clippingOffsetFactor,
-      dy: (b[0] / diagonal) * clippingOffsetFactor,
-      dz: (-1 / 2) * clippingOffsetFactor,
-      upX: 0,
-      upY: 0,
-      upZ: -1,
-      l: squareCenterX - squareWidth / 2,
-      r: squareCenterX + squareWidth / 2,
-      t: squareCenterY + height / 2,
-      b: squareCenterY - height / 2,
-    };
+  // Factor to reduce the clipping offset of the z coordinate to get a better angle on the dataset.
+  const zReductionFactor = 0.7;
```

Comment:
Is this needed because of the different z-scale in relation to x and y? If so, shouldn't it be computed from the dataset scale automatically as dataset scales differ widely?

Reply:
I thought of this as a way to simply change the view angle at the dataset. Here are two examples:
With `zReductionFactor = 1`:
![image](https://user-images.githubusercontent.com/39529669/89534561-69c99780-d7f5-11ea-8aef-1c7d81d406b8.png)
 With `zReductionFactor = 0.7`:
![image](https://user-images.githubusercontent.com/39529669/89534617-849c0c00-d7f5-11ea-8c6e-4a4e8fbcbe37.png)
As you can see: The angle changed.

But if you compare these I think the first version is fine too and thus the factor can be omitted.